src/Header/AcceptHeaderItem.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * Spiral Framework.
+ *
+ * @license   MIT
+ * @author    Pavel Z
+ */
+
+declare(strict_types=1);
+
+namespace Spiral\Http\Header;
+
+/**
+ * Represents "Accept" header single item.
+ *
+ * Can be used for comparing each item weight or constructing the "Accept" headers.
+ */
+class AcceptHeaderItem
+{
+    /** @var string|null */
+    private $value;
+
+    /** @var float */
+    private $quality;
+
+    /** @var array */
+    private $params;
+
+    /**
+     * Parse accept header string.
+     *
+     * @param string $string
+     * @return static
+     */
+    public static function fromString(string $string): self
+    {
+        $elements = explode(';', $string);
+
+        $mime = trim((string) array_shift($elements));
+        $quality = 1.0;
+        $params = [];
+
+        foreach ($elements as $element) {
+            $parsed = explode('=', trim($element), 2);
+
+            // Wrong params must be ignored
+            if (count($parsed) !== 2) {
+                continue;
+            }
+
+            $name = trim($parsed[0]);
+            $value = trim($parsed[1]);
+
+            if ($name === 'q') {
+                $quality = floatval($value);
+            } else {
+                $params[$name] = $value;
+            }
+        }
+
+        return new static($mime, $quality, $params);
+    }
+
+    /**
+     * Compare to header items, witch one is preferable.
+     * Return 1 if first value preferable or -1 if second, 0 in case of same weight.
+     *
+     * @param AcceptHeaderItem|string $a
+     * @param AcceptHeaderItem|string $b
+     * @return int
+     */
+    public static function compare($a, $b): int
+    {
+        $a = $a instanceof AcceptHeaderItem ? $a : AcceptHeaderItem::fromString((string) $a);
```

Comment:
What if someone will pass non-scalar value here? I assume if both are not comparable we can return 0, if only one is not comparable, then count it as a less value